src/sentry/migrations/0384_backfill_installation_ids.py

```diff
@@ -0,0 +1,63 @@
+# Generated by Django 2.2.28 on 2023-03-09 23:03
+
+from django.db import migrations
+
+from sentry.monitors.models import MonitorStatus
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+TERMINAL_STATES = [MonitorStatus.PENDING_DELETION, MonitorStatus.DELETION_IN_PROGRESS]
+DEFAULT_ENVIRONMENT_NAME = "production"
+
+
+def backfill_installation_ids(apps, schema_editor):
+    ServiceHook = apps.get_model("sentry", "ServiceHook")
+
+    # The actor id for all ServiceHooks where a project id is not specified is the installation id,
+    # but other actor ids are user ids.
+    queryset = RangeQuerySetWrapperWithProgressBar(
+        ServiceHook.objects.filter(project_id__isnull=True).values_list(
+            "id",
+            "actor_id",
+        ),
+        result_value_getter=lambda item: item[0],
+    )
+
+    for service_hook_id, actor_id in queryset:
+        try:
+            hook = ServiceHook.objects.get(id=service_hook_id)
+        except ServiceHook.DoesNotExist:
+            # Could happen from concurrent deletes, not a big deal.
+            continue
+        hook.installation_id = actor_id
+        hook.save()
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = True
```

Comment:
Since this is only 10-11k rows I think you could mark it as not dangerous and let it run with the deploy.